### ISSUE TITLE
Add unit tests for datanode side position management

### ISSUE THREAD

**Issue by peterbarrow** [vegaprotocol/vega#8449]
The core side has a set of tests that ensure the position engine works correctly. We should port those over to the datanode side of things to make sure we correctly build up the correct results on this side.

Here are the originals:
core/plugins/position_spec_test.go

Important thing to remember when writing the tests, all the events except the trade event has the prices given with market scaling, the trade events use position scaling.

**Comment by EVODelavega** [comment]
From the looks of things, the tests we have in (the old) core/integration/position_spec_test.go file have been ported over to datanode/entities/position_test.go

I also don't think there's any price data that uses market decimals. When orders are placed, the price fields immediately get scaled to asset decimals, and events all use the correct (asset) price data. The position decimals then have to be used to preserve the correct size-price ratio.

TL;DR: I believe this is all covered

**Comment by peterbarrow** [comment]
I believe some other work was done around this area resulting in decimal places being standard across all events.
Happy for this to be closed.
